sklearn/utils/estimator_checks.py

```diff
@@ -327,6 +332,39 @@ def _is_32bit():
     return struct.calcsize('P') * 8 == 32
 
 
+def check_estimators_are_deterministic(name, Estimator):
+    # all estimators should be deterministic with a fixed random_state
+    X, y = make_classification(n_classes=3, n_informative=3, random_state=3)
+    # We need to make sure that we have non negative data, for some estimators
+    X -= X.min() - .1
+
+    # XXX skipping bad estimators for the moment
+    if name in ('HuberRegressor', 'LogisticRegressionCV',
+                'LinearRegression', 'RANSACRegressor',
+                'RadiusNeighborsClassifier'):
+        return
+
+    X_train, X_test, y_train, y_test = train_test_split(X, y,
+                                                        train_size=0.5,
+                                                        random_state=3)
+    if "MultiTask" in name:
+        y_train = np.reshape(y_train, (-1, 1))
+        y_test = np.reshape(y_test, (-1, 1))
+
+    needs_state = 'random_state' in signature(Estimator.__init__).parameters
```

Comment:
You end up with something like `needs_state = 'random_state' in Estimator().get_params()` is that really nicer than inspecting the `__init__` arguments?